packages/runtime/src/lesson-files.ts

```diff
@@ -128,3 +156,7 @@ function convertToFiles(json: Record<string, string | { base64: string }>): File
 
   return result;
 }
+
+function fromFilesRefToType(filesRef: string): 'template' | 'files' | 'solution' {
+  return filesRef.startsWith('template-') ? 'template' : filesRef.endsWith('files.json') ? 'files' : 'solution';
```

Comment:
Now that it's not inline anymore, I'd rather if a `if-else` statement here with early returns?

```ts
if (filesRef.startsWith('template-') {
  return 'template';
}

if (filesRef.endsWith('files.json') {
  return 'files';
}

return 'solution';
```

It's a bit longer, but much more readable (and maintainable) IMO.